src/plugins/vis_type_timelion/public/components/timelion_vis_component.tsx

```diff
@@ -393,32 +174,79 @@ function TimelionVisComponent({
     [fireEvent]
   );
 
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plotselected').on('plotselected', plotSelectedHandler);
-    }
-  }, [chartElem, plotSelectedHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('mouseleave').on('mouseleave', mouseLeaveHandler);
-    }
-  }, [chartElem, mouseLeaveHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plothover').on('plothover', plotHoverHandler);
-    }
-  }, [chartElem, plotHoverHandler]);
+  const onRenderChange = useCallback(
+    (isRendered: boolean) => {
+      if (isRendered) {
+        renderComplete();
+      }
+    },
+    [renderComplete]
+  );
 
   const title: string = useMemo(() => last(compact(map(seriesList.list, '_title'))) || '', [
     seriesList.list,
   ]);
 
+  const tickFormat = useMemo(
+    () => createTickFormat(interval, kibana.services.timefilter, kibana.services.uiSettings),
+    [interval, kibana.services.timefilter, kibana.services.uiSettings]
+  );
+
   return (
-    <div ref={elementRef} className="timChart">
-      <div className="chart-top-title">{title}</div>
-      <div ref={canvasRef} className="chart-canvas" />
+    <div className="timelionChart">
+      <div className="timelionChart__topTitle">{title}</div>
+      <Chart ref={chartRef as RefObject<Chart>} renderer="canvas" size={{ width: '100%' }}>
+        <Settings
+          onBrushEnd={brushEndListener}
+          showLegend
+          legendPosition={getLegendPosition()}
+          onRenderChange={onRenderChange}
+          onPointerUpdate={handleCursorUpdate}
+          theme={kibana.services.chartTheme.useChartsTheme()}
+          baseTheme={kibana.services.chartTheme.useChartsBaseTheme()}
+          tooltip={{
+            snap: true,
+            type: TooltipType.VerticalCursor,
+          }}
+          externalPointerEvents={{ tooltip: { visible: false } }}
+        />
+        <Axis id="bottom" position={Position.Bottom} showOverlappingTicks tickFormat={tickFormat} />
+        {chart[0]._global?.yaxes ? (
+          chart[0]._global.yaxes.map((axis: IAxis, index: number) => {
+            return (
+              <Axis
+                key={index}
+                id={axis.position + axis.axisLabel}
+                title={axis.axisLabel}
+                position={axis.position}
+                tickFormat={axis.tickFormatter}
+                gridLine={{
+                  stroke: GRID_LINE_STROKE,
+                  visible: true,
+                }}
```

Comment:
The chartTheme already have the gridLine style, we should not overwrite it:
https://github.com/elastic/eui/blob/75118102c766c1010682f7aceeb4b17347fa35e6/src/themes/charts/themes.ts#L151-L167

Reply:
removed